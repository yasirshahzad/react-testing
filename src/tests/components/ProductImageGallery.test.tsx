import { render, screen } from "@testing-library/react";
import ProductImageGallery from "../../components/ProductImageGallery";

describe("ProductImageGallery", () => {
  const imageUrls = [
    "https://example.com/image1.jpg",
    "https://example.com/image2.jpg",
  ];

  it("should not render when imageUrls is empty", () => {
    const { container } = render(<ProductImageGallery imageUrls={[]} />);

    expect(container).toBeEmptyDOMElement();
  });

  it("should render the image gallery", () => {
    render(<ProductImageGallery imageUrls={imageUrls} />);

    const list = screen.getAllByRole("img");

    list.forEach((image, index) => {
      expect(image).toBeInTheDocument();
      expect(image).toHaveAttribute("src", imageUrls[index]);
    });
  });
});
