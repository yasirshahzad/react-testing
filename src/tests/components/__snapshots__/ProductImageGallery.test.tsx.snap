// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductImageGallery > should render the image gallery 1`] = `
<div>
  <ul>
    <li>
      <img
        src="https://example.com/image1.jpg"
      />
    </li>
    <li>
      <img
        src="https://example.com/image2.jpg"
      />
    </li>
  </ul>
</div>
`;
